package/src/lib/SpotlightTour.provider.tsx

```diff
@@ -128,7 +147,24 @@ export const SpotlightTourProvider = forwardRef<SpotlightTour, SpotlightTourProv
 
   const start = useCallback((): void => {
     renderStep(0);
-  }, [renderStep]);
+    onStart?.();
+  }, [renderStep, onStart]);
+
+  const startOnce = useCallback(() => {
+    if (!tourId) {
+      setTourId(hash(steps));
+      renderStep(0);
+      onStart?.();
+    }
+  }, [renderStep, onStart, steps]);
+
+  useEffect(() => {
+    if (autoStart === "always") {
+      start();
+    } else if (autoStart === "once") {
+      startOnce();
+    }
```

Comment:
Let's use a switch statement to avoid the if-else and use a safe pattern matching instead 🙂